jax/_src/pallas/mosaic/lowering.py

```diff
@@ -2267,6 +2267,47 @@ def _shift_right_logical_lowering_rules(ctx: LoweringRuleContext, x, d):
 skip_mlir_conversions.add(lax.shift_right_logical_p)
 
 
+# based on https://github.com/openxla/xla/blob/a7a09d56c3599123f8148bbf3e44c9ebc04624b9/xla/mlir_hlo/mhlo/transforms/chlo_legalize_to_hlo/chlo_legalize_to_hlo.cc#L644-L802
+def _erf_inv_32_helper(x):
+  k_degree = 9
+  w_less_than_5_constants = [
+    2.81022636e-08,  3.43273939e-07, -3.5233877e-06,
+    -4.39150654e-06, 0.00021858087,  -0.00125372503,
+    -0.00417768164,  0.246640727,    1.50140941,
+  ]
+  w_greater_than_5_constants = [
+    -0.000200214257, 0.000100950558, 0.00134934322,
+    -0.00367342844,  0.00573950773,  -0.0076224613,
+    0.00943887047,   1.00167406,     2.83297682,
+  ]
+
+  w = -jnp.log1p(x * -x)
+  lt = w < 5.0
+
+  w = jnp.where(lt, w - 2.5, jnp.sqrt(w) - 3.0)
+
+  p = jnp.where(lt, w_less_than_5_constants[0], w_greater_than_5_constants[0])
+  for i in range(1, k_degree):
+    c = jnp.where(lt, w_less_than_5_constants[i], w_greater_than_5_constants[i])
+    p = c + p * w
+
+  result = p * x
```

Comment:
Nit: I would personally inline both `p * x` and `edge_cases`.

Reply:
Done